### PR TITLE
Only send required fields when doing mergefield create or patch

### PR DESCRIPTION
**Pull request by JonasSteur**
https://vikingco.atlassian.net/browse/INT-5393

By setting the `only` attribute on a marshmallow schema we can limit the
fields that get sent by a request. Before the python 3.7 upgrade it wasn't needed
to set this explicitly as empty fields would not be sent. Now empty
fields will be send with a NULL value which causes
errors.(https://sentry.unleashed.be/unleashed/budskap/issues/283282)

https://developer.mailchimp.com/documentation/mailchimp/reference/lists/merge-fields
Budskap will always send a name, type and tag.

### CODE REVIEW

File: wingmonkey/merge_fields.py [hunk 1]
```diff
@@ -29,7 +29,9 @@ def create(self, list_id, merge_field_instance):
         :param merge_field_instance: MergeField
         :return: MergeField instance created on server
         """
+        self.only = ('tag', 'name', 'type')
         response = self.session.post(f'lists/{list_id}/merge-fields', json=self.dumps(merge_field_instance))
+        self.only = ()
```
Comment (freeduerinckx) [inline wingmonkey/merge_fields.py]: What if the response throws an error? Wouldn't it be better to make it as a context manager?
Another question, why don't we only send those fields that we want to merge?

File: wingmonkey/tests/test_merge_fields.py [hunk 2]
```diff
@@ -73,7 +73,7 @@ def test_merge_field_with_tag_not_str(expected_merge_field):
 
 
 def test_merge_field_create(expected_merge_field):
-    merge_field = MergeField(**expected_merge_field)
+    merge_field = MergeField(name='Firestarter', tag='FIRSRTR', type=MergeFieldTypes.TEXT, list_id='TheFatOfTheLand')
```
Comment (freeduerinckx) [inline wingmonkey/tests/test_merge_fields.py]: Or I didn't understood the change or something is wrong. If you are sending those 4 fields, why do you expected a lot more values? Why is list_id sent, it's not part of parameters of the `only` call?

File: wingmonkey/tests/test_merge_fields.py [hunk 3]
```diff
@@ -92,7 +92,8 @@ def test_merge_field_read(expected_merge_field):
 
 
 def test_merge_field_update(expected_merge_field):
-    merge_field = MergeField(**expected_merge_field)
+    merge_field = MergeField(name='Firestarter', tag='FIRSRTR', type=MergeFieldTypes.TEXT, list_id='TheFatOfTheLand',
```
Comment (freeduerinckx) [inline wingmonkey/tests/test_merge_fields.py]: Or I didn't understood the change or something is wrong. If you are sending those 5 fields, why do you expected a lot more values? Why is list_id sent, it's not part of parameters of the `only` call?

**Comment by JonasSteur** [comment]
@freeduerinckx fixup pushed. I found a nicer way to do this and in the meantime cleaned up the tests a bit to provide more realistic responses and check if ValidationErrors get raised when expected

File: wingmonkey/merge_fields.py [hunk 4]
```diff
@@ -15,18 +15,23 @@ class Meta:
...
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.dump_only = ('_links',)
 
     def create(self, list_id, merge_field_instance):
         """
         :param list_id: id of list the merge field will be defined for
         :param merge_field_instance: MergeField
+        :param only: Tuple;  limit fields to be sent to these fields
```
Comment (freeduerinckx) [inline wingmonkey/merge_fields.py]: Remove this line

**Comment by JonasSteur** [comment]
@freeduerinckx fixup pushed to expand tests and verify the request json

**Pull request merged**